src/note.rs

```diff
@@ -282,28 +340,78 @@ impl Note {
 
     /// Derives the nullifier for this note.
     pub fn nullifier(&self, fvk: &FullViewingKey) -> Nullifier {
+        let selected_rseed = self.rseed_split_note.unwrap_or(self.rseed);
+
         Nullifier::derive(
             fvk.nk(),
             self.rho.0,
-            self.rseed.psi(&self.rho),
+            selected_rseed.psi(&self.rho),
             self.commitment(),
+            self.rseed_split_note.is_some(),
         )
     }
+
+    /// Create a split note which has the same values than the input note except for
+    /// `rseed_split_note` which is equal to a random seed.
```

Comment:
If my previous comment is actioned, then `rseed_split_note` won't be anywhere in the public API of `Note`, so this documentation should be reworded.

Additionally, this method should only be valid to call on Custom Assets (for native assets, dummy notes should be used instead). Given that this method is rather low-level, I think it's fine to panic on incorrect usage rather than returning `Option<Self>`.

```suggestion
    /// Creates a [Split Input note] from a Custom Asset note, for use on the Spend side
    /// of an Output-only Action.
    ///
    /// # Panics
    ///
    /// Panics if `self.asset().is_native()`.
    ///
    /// [Split Input note]: https://zips.z.cash/zip-0226#split-notes
```

I'm also wondering whether this method should be public or not. We need this when preparing the final set of spends and outputs within the Action (roughly encompassed by the Constructor and IO Finalizer roles in PCZTs), and presumably this can currently all be handled internally to `Builder`?